rust/mdb_shard/src/shard_file_manager.rs

```diff
@@ -129,23 +164,76 @@ impl ShardFileManager {
         new_shards: &[MDBShardFile],
         shards_are_permanent: bool,
     ) -> Result<()> {
-        let mut current_lookup = self.shard_file_lookup.write().await;
-
-        for s in new_shards {
+        let mut shards_lg = self.shard_file_lookup.write().await;
+
+        // Go through and register the shards in order of newest to oldest
+        let mut new_shards: Vec<(&MDBShardFile, SystemTime)> = new_shards
+            .iter()
+            .map(|s| {
+                let modified = std::fs::metadata(&s.path)?.modified()?;
```

Comment:
It's very difficult to predict which data file a user wants to change. When the loading order actually matters, i.e. when number of shards or number of chunks cross the threshold, a user may suddenly encounter degraded performance when changing a part of an old big file.
I lean towards loading in a random order. WDYT?

Reply:
It will always succeed, but since it's dedup, I think that it's more likely that content gets deduplicated against a recent shard than an older one.  Also, with the current configuration, the first 16 TB or so of chunk content is held in memory, so this only makes sense on much, much larger repos.   I'd say since it always succeeds, it's just an optimization, but most recent is more likely than other rules. .

Reply:
That's fair.